File: 26.cpp
class Solution
{
public:
    int removeDuplicates(vector<int> &nums)
    {
        int len = nums.size();
        if (len <= 1)
            return len;
        int count = 1;
        for (int i = 1; i < len; i++)
        {
            if (nums[i] != nums[i - 1])
            {
                nums[count] = nums[i];
                count++;
            }
        }
        return count;
    }
};

File: 26.scala
object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
      val length = nums.length
      if (length <= 1) return length

      var count = 1
      var (l, r) = (0, 0)
      for (i <- (1 until length)) {
        if (nums(i) != nums(i - 1)) {
          nums(count) = nums(i)
          count += 1
        }
      }
      count
    }
}